internal/subscriber/model.go

```diff
@@ -151,14 +152,29 @@ func (a *AckMessage) HasHitDeadline() bool {
 }
 
 // ParseAckID ...
-func ParseAckID(ackID string) *AckMessage {
+func ParseAckID(ackID string) (*AckMessage, error) {
 	// split the message and parse tokens
 	parts := strings.Split(ackID, ackIDSeparator)
+	if !isValidAckID(parts) {
+		return nil, merror.Newf(merror.InvalidArgument, "AckID received is not in expected format")
+	}
+	partition, err := strconv.ParseInt(utils.Decode(parts[3]), 10, 0)
+	if err != nil {
+		fmt.Println(err)
+		return nil, fmt.Errorf("error in parsing partition value %v to int", utils.Decode(parts[3]))
+	}
+
+	offset, err := strconv.ParseInt(utils.Decode(parts[4]), 10, 0)
 
-	// TODO : add validations
-	partition, _ := strconv.ParseInt(utils.Decode(parts[3]), 10, 0)
-	offset, _ := strconv.ParseInt(utils.Decode(parts[4]), 10, 0)
-	deadline, _ := strconv.ParseInt(utils.Decode(parts[5]), 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("error in parsing offset value %v to int", utils.Decode(parts[4]))
```

Comment:
decode once, store in var and re-use the same. Do for all parts.

Reply:
Done